src/nvmsgbroker_pipeline.py

```
@@ -0,0 +1,147 @@
+            client.disconnect()
+            print(f" Sent frame {frame_meta.frame_num} to MQTT")
+        except Exception as e:
+            print(f" MQTT Error: {e}")
+        
+        l_frame = l_frame.next
+    
+    return Gst.PadProbeReturn.OK
+
+def build_pipeline():
+    pipeline = Gst.Pipeline.new("direct-mqtt-pipeline")
+
+    # Source elements
+    rtspsrc = Gst.ElementFactory.make("rtspsrc", "source")
+    depay = Gst.ElementFactory.make("rtph264depay", "depay")
+    parse = Gst.ElementFactory.make("h264parse", "parse")
+    decode = Gst.ElementFactory.make("decodebin", "decode")
+    convert = Gst.ElementFactory.make("videoconvert", "convert")
+    nvvideoconvert = Gst.ElementFactory.make("nvvideoconvert", "nvvideoconvert")
+    
+    # DeepStream elements
+    streammux = Gst.ElementFactory.make("nvstreammux", "streammux")
+    nvinfer = Gst.ElementFactory.make("nvinfer", "nvinfer")
+    fakesink = Gst.ElementFactory.make("fakesink", "sink")
+    
+    # Add elements to pipeline
+    elements = [rtspsrc, depay, parse, decode, convert, nvvideoconvert, 
+                streammux, nvinfer, fakesink]
+    for element in elements:
+        pipeline.add(element)
+
+    # Configure properties
+    rtspsrc.set_property("location", RTSP_URL)
+    
+    streammux.set_property("batch-size", 1)
+    streammux.set_property("width", 640)
+    streammux.set_property("height", 480)
+    
+    nvinfer.set_property("config-file-path", CONFIG_FILE)
+
+    # הוספת probe ל-nvinfer
+    nvinfer_src_pad = nvinfer.get_static_pad("src")
+    nvinfer_src_pad.add_probe(Gst.PadProbeType.BUFFER, direct_mqtt_probe, 0)
+
+    # Link static elements
+    convert.link(nvvideoconvert)
+    streammux.link(nvinfer)
+    nvinfer.link(fakesink)
+
+    # Dynamic pad connections
+    def on_pad_added_rtspsrc(src, pad):
+        sinkpad = depay.get_static_pad("sink")
+        if not sinkpad.is_linked():
+            pad.link(sinkpad)
```

Linking dynamic pads without verifying their capabilities (`caps`) may lead to silent failures or format mismatches that are difficult to debug.

Therefore, it is recommended to inspect or log the pad's capabilities before calling `pad.link(sinkpad)` to ensure compatibility.